### PR TITLE
feat(private-database.upgrade): deprecate order/hosting routes

### PR DESCRIPTION
**Pull request by lucas-chaigne**
Signed-off-by: Lucas Chaigne <lucas.chaigne@ovhcloud.com>

| Question         | Answer
| ---------------- | ---
| Branch?          | develop
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | WEB-12807
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] Breaking change is mentioned in relevant commits

## Description

APIv6 routes order/hosting are deprecated. This PR aim is to switch order/hosting/privateSQL route to /order/upgrade/cloudDB for cloudDB and privateSQL upgrade pages.

### CODE REVIEW

File: packages/manager/apps/web/client/app/private-database/database/ram/update/private-database-database-ram-update.controller.js [hunk 1]
```diff
@@ -96,39 +87,31 @@ angular.module('App').controller(
...
-          angular.noop,
-          (duration) => {
-            this.data.durations.available = duration;
-          },
-        );
+        .getUpgradeProvisionalOrder(this.productId, this.model.planCode)
+        .then((response) => {
+          this.loading.provisionalOrder = false;
+          this.data.provisionalOrder = response;
+        });
```
Comment (y4nnL) [inline packages/manager/apps/web/client/app/private-database/database/ram/update/private-database-database-ram-update.controller.js:98]: There should be a `catch` statement that alerts something went wrong when fetching
As a result `this.loading.provisionalOrder = false;` should move into a `finally` statement

File: packages/manager/apps/web/client/app/private-database/database/ram/update/private-database-database-ram-update.html [hunk 2]
```diff
@@ -19,9 +19,9 @@
                         class="oui-select__input"
                         name="capacity"
                         id="selectRamCapacity"
-                        data-ng-disabled="ramCtrl.loading.availableRam"
-                        data-ng-model="ramCtrl.model.capacity"
-                        data-ng-options="ram + ('unit_size_MB' | translate) for ram in ramCtrl.data.availableRam | orderBy:sortRam"
+                        data-ng-disabled="ramCtrl.loading.availablePlans"
+                        data-ng-model="ramCtrl.model.planCode"
+                        data-ng-options="ramCtrl.constructor.getCapacityFromPlanCode(planCode) + ('unit_size_MB' | translate) for planCode in ramCtrl.data.availablePlans | orderBy:sortRam"
```
Comment (y4nnL) [inline packages/manager/apps/web/client/app/private-database/database/ram/update/private-database-database-ram-update.html:24]:
```suggestion
                        data-ng-options="ramCtrl.getCapacityFromPlanCode(planCode) + ('unit_size_MB' | translate) for planCode in ramCtrl.data.availablePlans | orderBy:sortRam"
```

File: packages/manager/apps/web/client/app/private-database/database/ram/update/private-database-database-ram-update.controller.js [hunk 3]
```diff
@@ -168,8 +147,8 @@ angular.module('App').controller(
         });
     }
 
-    static isProrataDuration({ duration }) {
-      return /^upto/.test(duration);
+    static getCapacityFromPlanCode(planCode) {
```
Comment (y4nnL) [inline packages/manager/apps/web/client/app/private-database/database/ram/update/private-database-database-ram-update.controller.js:150]: The static keyword is not necessary here
```suggestion
    getCapacityFromPlanCode(planCode) {
```

File: packages/manager/apps/web/client/app/private-database/database/ram/update/private-database-database-ram-update.html [hunk 4]
```diff
@@ -46,55 +46,25 @@
...
-                                    data-ng-bind-html="duration.prices | price: ramCtrl.data.ovhSubsidiary"
-                                ></span>
-                            </span>
-                        </span>
-                    </label>
-                </div>
-            </form>
+            <div data-ng-if="!ramCtrl.loading.provisionalOrder">
+                <span data-translate="privateDatabase_RAM_add_duration"> </span>
+                &nbsp;:
```
Comment (y4nnL) [inline packages/manager/apps/web/client/app/private-database/database/ram/update/private-database-database-ram-update.html:63]: Prefer the use of utility classes
```suggestion
                <span class="pr-2" data-translate="privateDatabase_RAM_add_duration"></span>
```

File: packages/manager/apps/web/client/app/private-database/state/private-database-state.controller.js [hunk 5]
```diff
@@ -66,9 +66,13 @@ export default class PrivateDatabaseStateCtrl {
...
-    this.privateDatabaseService.canOrderRam(this.productId).then((canOrder) => {
-      this.canOrderRam = canOrder;
-    });
+    this.privateDatabaseService
+      .getUpgradePlans(this.productId)
+      .then((response) => {
+        if (response.length > 0) {
+          this.canOrderRam = true;
+        }
+      });
```
Comment (y4nnL) [inline packages/manager/apps/web/client/app/private-database/state/private-database-state.controller.js:75]: Code duplication
https://github.com/ovh/manager/pull/7899/files#diff-47f38362ec6da4c315e1c44e3508148b0c8ff06e9868c792243ccf4fa5d6f5a8R43
You should refactor this inside the private database service
```suggestion
    this.privateDatabaseService
      .canOrderRam(this.productId)
      .then((canOrderRam) => {
          this.canOrderRam = canOrderRam;
      });
```

**Comment by lucas-chaigne** [comment]
Pull request is closed because after testing edge cases we found out that routes routes GET /${serviceName}/ram et /${serviceName}/ram/${duration} and POST /${serviceName}/ram/${duration} cant be deprecated (due to legacy billing)
I'll do an other PR on branch dev/lucas.chaigne/WEB-12807-order-db to remove other deprecated routes

**Pull request closed without merging**